internal/api/validate.go

```diff
@@ -93,6 +94,43 @@ func NewValidator() *validator.Validate {
 		panic(err)
 	}
 
+	// Use this for string fields that must be a valid Kubernetes qualified name.
+	err = validate.RegisterValidation("k8s_qualified_name", func(fl validator.FieldLevel) bool {
+		field := fl.Field()
+		if field.Kind() != reflect.String {
+			panic("String type required for k8s_qualified_name")
+		}
+		return len(k8svalidation.IsQualifiedName(field.String())) == 0
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	// Use this for string fields that must be a valid Kubernetes label value.
+	err = validate.RegisterValidation("k8s_label_value", func(fl validator.FieldLevel) bool {
+		field := fl.Field()
+		if field.Kind() != reflect.String {
+			panic("String type required for k8s_label_value")
+		}
+		return len(k8svalidation.IsValidLabelValue(field.String())) == 0
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	// Use this for version ID fields that might begin with "openshift-v".
+	err = validate.RegisterValidation("openshift_version", func(fl validator.FieldLevel) bool {
+		field := fl.Field()
+		if field.Kind() != reflect.String {
+			panic("String type required for openshift_version")
+		}
+		_, err := NewOpenShiftVersion(field.String())
+		return err == nil
+	})
+	if err != nil {
+		panic(err)
```

Comment:
Should we be panic'ing on error?  

If so, are we handling goroutines panicking not taking down he whole app?

Reply:
> Should we be panic'ing on error?

We're panic'ing on a registration error, which will happen on startup either 100% of the time or never, kinda like calling `regex.MustCompile`.  In this particular case, the only way the registration fails is if you pass nonsense arguments like a `nil` function.

> If so, are we handling goroutines panicking not taking down he whole app?

More generally, yes we do have middleware in the frontend to catch a panic.